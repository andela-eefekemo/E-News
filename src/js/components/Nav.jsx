/* global location localStorage document $*/
import React from 'react';
import { IndexLink, Link, browserHistory } from 'react-router';
import PropTypes from 'prop-types';
/**
 * @class Nav
 * @extends {React.Component}
 */
class Nav extends React.Component {
  constructor() {
    super();
    this.logout = this.logout.bind(this);
  }
  /**
   * Logs out user from website
   * @param {event} e Takes in onClick event
   * @return {void}
   */
  logout(e) {
    e.preventDefault();
    localStorage.removeItem('User');
    browserHistory.replace('/');
    location.reload();
  }
  render() {
    return (
      <div>
        <nav>
          <div className="nav-wrapper teal lighten-1">
            <a href="#!" className="brand-logo logo">E-News</a>
            <ul className="right hide-on-med-and-down">
              <li><IndexLink to="/">Home</IndexLink></li>

              <li><Link to="/logout" onClick={this.logout}>Logout</Link></li>
            </ul>
            <ul className="side-nav" id="mobile-demo">
              <li><Link to="/">Home</Link></li>
              <li className="divider" />
              <li> <Link to="/logout" onClick={this.logout}>Logout</Link></li>
            </ul>
          </div>
        </nav>
      </div>
    );
  }
}

// Set Default Props
Nav.defaultProps = {
  info: {
    imageURL: '',
    name: ''
  }
};
// Set Props
Nav.propTypes = {
  info: PropTypes.object,
  imageURL: PropTypes.string,
  name: PropTypes.string
};

export default Nav;
