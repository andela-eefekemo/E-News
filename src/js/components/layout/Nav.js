import React from 'react';
import { IndexLink, Link } from 'react-router';

export default class Nav extends React.Component {
  render() {
    return (
      <div>
        <nav>
          <div className="nav-wrapper teal lighten-1">
            <div data-activates="mobile-demo" className="button-collapse">
              <i className="material-icons">menu</i>
            </div>
            <ul className="right hide-on-med-and-down">
              <li><IndexLink to="/">Home</IndexLink></li>
              <li><Link to="/headlines">Headlines</Link></li>
            </ul>
            <ul className="side-nav" id="mobile-demo">
              <li><IndexLink to="/">Dashboard</IndexLink></li>
              <li><Link to="/headlines">Headlines</Link></li>
            </ul>
          </div>
        </nav>
      </div>
    );
  }
}

