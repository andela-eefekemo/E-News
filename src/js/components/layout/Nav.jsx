/* global location localStorage*/
import React from 'react';
import { IndexLink, Link } from 'react-router';

// Nav Component
const Nav = () => {
  /**
   * Logs out user from website
   * @param {event} e Takes in onClick event
   */
  const logout = (e) => {
    e.preventDefault();
    localStorage.removeItem('User');
    location.reload();
  };
  return (
    <div>
      <nav>
        <div className="nav-wrapper teal lighten-1">
          <div data-activates="mobile-demo" className="button-collapse">
            <i className="material-icons">menu</i>
          </div>
          <ul className="right hide-on-med-and-down">
            <li><IndexLink to="/">Home</IndexLink></li>
            <li ><Link to="/logout" onClick={logout}>Logout</Link></li>
          </ul>
          <ul className="side-nav" id="mobile-demo">
            <li><Link to="/">Home</Link></li>
            <li> <Link to="/logout" onClick={logout}>Logout</Link></li>
          </ul>
        </div>
      </nav>
    </div>
  );
};

export default Nav;
