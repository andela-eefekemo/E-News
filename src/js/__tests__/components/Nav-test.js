import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Nav from '../../components/layout/Nav';


test('Navbar component', () => {
  
  const userProfile = {
    name:'Eguono',
    email:'john.doe'
  }
  localStorage.setItem('User', userProfile);
  const component = mount(<Nav />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
