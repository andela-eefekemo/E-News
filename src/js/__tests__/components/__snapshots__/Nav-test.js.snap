// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component 1`] = `
<Nav>
  <div>
    <nav>
      <div
        className="nav-wrapper teal lighten-1"
      >
        <div
          className="button-collapse"
          data-activates="mobile-demo"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </div>
        <ul
          className="right hide-on-med-and-down"
        >
          <li>
            <IndexLink
              to="/"
            >
              <Link
                onlyActiveOnIndex={true}
                style={Object {}}
                to="/"
              >
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  Home
                </a>
              </Link>
            </IndexLink>
          </li>
          <li>
            <Link
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/logout"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Logout
              </a>
            </Link>
          </li>
        </ul>
        <ul
          className="side-nav"
          id="mobile-demo"
        >
          <li>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Home
              </a>
            </Link>
          </li>
          <li>
             
            <Link
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/logout"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Logout
              </a>
            </Link>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</Nav>
`;
